### ISSUE TITLE
[24.10.0-rc2] mediatek: bpi-r4: wifi led problem

### ISSUE THREAD

**Issue by MichelHanzen** [openwrt/openwrt#17245]
### Describe the bug

![image](https://github.com/user-attachments/assets/2e52f2e2-5e1d-43fe-be71-436b4d91a54a)
2g/5g/6g blue leds are not working with openwrt 24.10.0-rc2

### OpenWrt version

r28161-ea17e958b9

### OpenWrt release

24.10.0-rc2

### OpenWrt target/subtarget

mediatek/filogic

### Device

Bananapi BPI-R4

### Image kind

Official downloaded image

### Steps to reproduce

configure leds with the Luci "System/LED configuration" menu
![image](https://github.com/user-attachments/assets/83ed6ba6-2bcc-43ae-89fe-74f5810c3229)

### Actual behaviour

2g/5g/6g blue leds are not working

### Expected behaviour

the thre blue LED's should blink when the relative wifi are up. It works with the BPI-R4-BE1350-WIFI_MP4_0-SDK-20240620 

### Additional info

Link to the BPI-R4-BE1350-WIFI_MP4_0-SDK-20240620 image:
https://drive.google.com/file/d/1U0tnvoaxDujO1q5QNxnw0i2d93X2zi9r/view?usp=sharing

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by MichelHanzen** [comment]
Problem not solved with 24.10.0-rc3

**Comment by MichelHanzen** [comment]
Same problem with 24.10.0-rc4. It looks like nobody cares...
Not taken i charge. I will not loose much time with testing. Back to BPI-R4-BE1350-WIFI_MP4_0-SDK-20240620 image

**Comment by sasha-rnd** [comment]
24.10.0-rc4 The problem is not solved

**Comment by MichelHanzen** [comment]
Not solved with 24.10.0-rc5

**Comment by EPinci** [comment]
Uhm, not authoritative but I think the led configuration you see there is for the leds in the LAN ports (back side, at the side of each lan port), not for the leds of the wifi nic (that might not necessarily be installed at all).
I haven't seen any support at all for the wifi board leds.

**Comment by MichelHanzen** [comment]
You're probably right but it's a pity that these front wifi leds are not managed by openwrt 24.10 though they are by BPI-R4-BE1350-WIFI_MP4_0-SDK-20240620 image

![Image](https://github.com/user-attachments/assets/d939cf31-ac37-4646-9434-4d7aae8787b5)

**Comment by EPinci** [comment]
Yeah, I know. But since those leds are actually on the wifi board, is probably on the driver/firmware to expose them.

**Comment by sasha-rnd** [comment]
24.10.0 r28427 The problem is not solved

**Comment by Rahzadan** [comment]
The 2G led can by manipulated using **mt76-phy0** in the LED configuration. Unfortunately **mt76-phy1** and **mt76-phy2** don't exist anymore because bediatek changed the way the physical Wifi interfaces work - they merged them all into a single PHY - whereas they used to be separated.

You can see it here in this diagram:

![Image](https://github.com/user-attachments/assets/a8e8d9c2-c12b-4305-bbc7-1a61dd856a02)

You can see the difference in "Before MLO" vs "After MLO". Once they implemented MLO, the 3 existing PHY interfaces (0, 1, 2) became one single phy0. 

I'm not sure how to fix this, but someone would need to somehow make the other 2 LEDs available in the dropdown as something like **mt76-phy0.0** **mt76-phy0.1** **mt76-phy0.0** as these are the names of the 3 wireless interfaces now.

**Comment by logiconcepts819** [comment]
I was looking at the series of MediaTek patches linked to [the mt76 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mt76/patches/0041-mtk-mt76-rework-statistics-report-for-mt7996.patch) shared in #21177 (specifically in [this comment](https://github.com/openwrt/openwrt/issues/21177#issuecomment-3696919802)), and I happened to stumble upon [this patch in mac80211](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mac80211/patches/subsys/0032-mtk-mac80211-export-ieee80211_tpt_led_trig_tx-rx-for.patch).  It appears to expose some methods in the mac80211 driver, `ieee80211_tpt_led_trig_tx` and `ieee80211_tpt_led_trig_rx`, for activating the Wi-Fi LEDs so that the mt76 driver can activate the appropriate LEDs directly.

Perhaps those two patches (i.e., [the mac80211 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mac80211/patches/subsys/0032-mtk-mac80211-export-ieee80211_tpt_led_trig_tx-rx-for.patch) and [the mt76 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mt76/patches/0041-mtk-mt76-rework-statistics-report-for-mt7996.patch)) could provide some clue for fixing this Wi-Fi LED problem.  Regardless, we probably need some additional changes besides those two patches for customization of the Wi-Fi LEDs from LuCI and the availability of the 2.4 GHz, 5 GHz, and 6 GHz Wi-Fi interfaces in the LED configuration dropdown menu.

This is just my two cents.

**Comment by dsbaha** [comment]
> I was looking at the series of MediaTek patches linked to [the mt76 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mt76/patches/0041-mtk-mt76-rework-statistics-report-for-mt7996.patch) shared in [#21177](https://github.com/openwrt/openwrt/issues/21177) (specifically in [this comment](https://github.com/openwrt/openwrt/issues/21177#issuecomment-3696919802)), and I happened to stumble upon [this patch in mac80211](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mac80211/patches/subsys/0032-mtk-mac80211-export-ieee80211_tpt_led_trig_tx-rx-for.patch). It appears to expose some methods in the mac80211 driver, `ieee80211_tpt_led_trig_tx` and `ieee80211_tpt_led_trig_rx`, for activating the Wi-Fi LEDs so that the mt76 driver can activate the appropriate LEDs directly.
> 
> Perhaps those two patches (i.e., [the mac80211 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mac80211/patches/subsys/0032-mtk-mac80211-export-ieee80211_tpt_led_trig_tx-rx-for.patch) and [the mt76 patch](https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/refs/heads/master/autobuild/unified/filogic/mac80211/25.12/files/package/kernel/mt76/patches/0041-mtk-mt76-rework-statistics-report-for-mt7996.patch)) could provide some clue for fixing this Wi-Fi LED problem. Regardless, we probably need some additional changes besides those two patches for customization of the Wi-Fi LEDs from LuCI and the availability of the 2.4 GHz, 5 GHz, and 6 GHz Wi-Fi interfaces in the LED configuration dropdown menu.
> 
> This is just my two cents.

Also, I'm finding a lot of missing definitions in the device tree (dtsi/dtso).  Those are also covered in the patches/link you mentioned.    I see references to pwm and leds in the patches, so may be related.

**Comment by eduardocalfaia** [comment]
Hi guys, any update here? I have the same issue
Thanks

**Comment by dm9bbadd4** [comment]
Any further updates on this?

**Comment by 66Ton99** [comment]
Last SNAPSHOT (r33329-557b092273) doesn't work either
